pkg/apis/application/v1alpha1/types.go

```diff
@@ -82,11 +82,12 @@ type ApplicationSpec struct {
 
 // ResourceIgnoreDifferences contains resource filter and list of json paths which should be ignored during comparison with live state.
 type ResourceIgnoreDifferences struct {
-	Group        string   `json:"group,omitempty" protobuf:"bytes,1,opt,name=group"`
-	Kind         string   `json:"kind" protobuf:"bytes,2,opt,name=kind"`
-	Name         string   `json:"name,omitempty" protobuf:"bytes,3,opt,name=name"`
-	Namespace    string   `json:"namespace,omitempty" protobuf:"bytes,4,opt,name=namespace"`
-	JSONPointers []string `json:"jsonPointers" protobuf:"bytes,5,opt,name=jsonPointers"`
+	Group             string   `json:"group,omitempty" protobuf:"bytes,1,opt,name=group"`
+	Kind              string   `json:"kind" protobuf:"bytes,2,opt,name=kind"`
+	Name              string   `json:"name,omitempty" protobuf:"bytes,3,opt,name=name"`
+	Namespace         string   `json:"namespace,omitempty" protobuf:"bytes,4,opt,name=namespace"`
+	JSONPointers      []string `json:"jsonPointers,omitempty" protobuf:"bytes,5,opt,name=jsonPointers"`
```

Comment:
Both `JSONPointers` and `JQPathExpressions` are optional, whereas `JSONPointers` was required in the past. It would be a breaking change for `JQPathExpressions` to be required, and since these fields should probably align, I think it makes sense to make them both optional.

It also probably makes sense that one of these fields is set. I believe enforcement of this would have to happen within argo, which means users would not have the kubernetes api request rejected. So, I'm not sure how this should be addressed.

Reply:
Agree, we've used the same approach in similar changes.